src/NClient.Tools/NClient.DotNetTool/FacadeGenerator.cs

```diff
@@ -23,25 +23,32 @@ public FacadeGenerator(INClientFacadeGenerator facadeGenerator, ILogger<FacadeGe
             _facadeGenerator = facadeGenerator;
         }
 
-        public async Task<string> GenerateAsync(FacadeOptions.GenerationOptions generationOptions, string specification)
+        public async Task<string> GenerateAsync(CommonGenerationOptions generationOptions, string specification)
         {
-            var serializerType = (generationOptions.UseSystemTextJson, generationOptions.UseNewtonsoftJson) switch
-            {
-                { UseSystemTextJson: false, UseNewtonsoftJson: false } => SerializeType.SystemJsonText,
-                { UseSystemTextJson: true, UseNewtonsoftJson: false } => SerializeType.SystemJsonText,
-                { UseSystemTextJson: false, UseNewtonsoftJson: true } => SerializeType.NewtonsoftJson,
-                var x => throw new NotSupportedException($"Not supported serializer configuration: {nameof(x.UseSystemTextJson)} = {x.UseSystemTextJson}; {nameof(x.UseNewtonsoftJson)} = {x.UseNewtonsoftJson}")
-            };
-            
             var generationSettings = new FacadeGenerationSettings(
-                generationOptions.FacadeName, 
+                name: (generationOptions as ClientOptions.GenerationOptions)?.ClientName 
+                ?? (generationOptions as FacadeOptions.GenerationOptions)?.FacadeName
+                ?? "{controller}", 
                 generationOptions.Namespace,
+                generateClients: (generationOptions as FacadeOptions.GenerationOptions)?.GenerateClients ?? false,
+                generateFacades: (generationOptions as ClientOptions.GenerationOptions)?.GenerateFacades ?? false,
                 generationOptions.UseModelValidationAttributes,
                 generationOptions.UseNullableReferenceTypes,
                 generationOptions.UseCancellationToken,
                 generationOptions.GenerateDtoTypes,
-                serializerType);
+                GetSerializerType(generationOptions.UseSystemTextJson, generationOptions.UseNewtonsoftJson));
             return await _facadeGenerator.GenerateAsync(specification, generationSettings);
```

Comment:
"await" is really necessary here?

Reply:
No, but it's easier to investigate errors with it, because the exception will be thrown in the method where the error actually occurred

Reply:
I don't think we should provide exception info to users, shouldn't? 
And what about performance?  Await statement produce state machine, so... non-zero price.

Reply:
The overhead for this is quite low. I even checked in the client's benchmark - the difference was not visible. For the console command, it doesn't matter at all

Reply:
Yes, maybe it's quite low, but still non-zero.  As far as I remember - we have some "issues" with a performance. Ok, right here this is irrelevant. I suggest create some rule for static analyze on similar cases and don't put on hold this.

Reply:
There is code smells for sonarcloud too :-)

Reply:
Ok, I'll remove await )

Reply:
Done